tests/OrasProject.Oras.Tests/examples/AttachReferrer.cs

```diff
@@ -10,53 +10,60 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-using OrasProject.Oras;
+#region Usage
 using OrasProject.Oras.Registry.Remote;
 using OrasProject.Oras.Registry;
 using OrasProject.Oras.Registry.Remote.Auth;
-using Moq;
 using Microsoft.Extensions.Caching.Memory;
 
-public class AttachReferrer
+namespace OrasProject.Oras.Tests.Examples;
+
+public static class AttachReferrer
 {
-    public async Task AttachReferrerAsync()
+    // This example demonstrates a basic implementation of attaching a referrer manifest to an existing subject manifest.
+    // Cancellation tokens and exception handling are omitted for simplicity.
+    public static async Task AttachReferrerAsync()
     {
-        #region Usage
-        // This example demonstrates how to attach a referrer to an existing manifest.
+        const string registry = "localhost:5000";
+        const string repository = "myrepo/test";
 
-        // Create a HttpClient instance to be used for making HTTP requests.
+        // Create a HttpClient instance for making HTTP requests.
         var httpClient = new HttpClient();
 
-        // Create a repository instance with the target registry.
-        var mockCredentialProvider = new Mock<ICredentialProvider>();
-        var memoryCache = new MemoryCache(new MemoryCacheOptions());
-        var repo = new Repository(new RepositoryOptions()
+        // Create a simple credential provider with static credentials.
+        var credentialProvider = new SingleRegistryCredentialProvider(registry, new Credential
         {
-            Reference = Reference.Parse("localhost:5000/test"),
-            Client = new Client(httpClient, mockCredentialProvider.Object, new Cache(memoryCache)),
+            Username = "username",
+            RefreshToken = "refresh_token"
```

Comment:
nit: You may type something like

```suggestion
            Username = "put_your_username_here",
            RefreshToken = "put_your_refresh_token_here"
```